src/components/favorite-button.js

```diff
@@ -0,0 +1,33 @@
+import favoriteFilledIcon from "../../images/favorite-icon-filled.png";
+import favoriteLinedIcon from "../../images/favorite-icon-lined.png";
+
+export function createFavoriteButton({ restaurant, onToggle }) {
+  const favoriteButton = document.createElement("button");
+  favoriteButton.className = "restaurant__favorite-button";
+
+  const favoriteIcon = createFavoriteIcon(restaurant.isFavorite);
+  favoriteButton.appendChild(favoriteIcon);
+
+  favoriteButton.addEventListener("click", (event) => {
+    event.stopPropagation();
+
+    restaurant.isFavorite = !restaurant.isFavorite;
+    changeFavoriteIcon({
+      favoriteIcon: favoriteIcon,
+      isFavorite: restaurant.isFavorite,
+    });
+    onToggle(restaurant.name);
+  });
+
+  return favoriteButton;
+}
+
+function createFavoriteIcon(isFavorite) {
+  const favoriteIcon = document.createElement("img");
+  changeFavoriteIcon({ favoriteIcon: favoriteIcon, isFavorite: isFavorite });
```

Comment:
왕소소하게 아래처럼 축약할 수도 있겠네용 (key와 value가 같으면 축약 가능)
```suggestion
  changeFavoriteIcon({ favoriteIcon, isFavorite });
```

Reply:
@yujo11 
key와 value가 같으면 생략이 가능하다는 점은 인지하고 있습니다!
다만, 어떤 key에 어떤 value를 넣을지 key를 명시하는 것이 좋은 것 같더라구요. 😄

Reply:
이런 관점에서도 생각해 볼 수 있겠군요! 👍 👍 👍